src/solvers/dgsem_tree/indicators_3d.jl

```diff
@@ -93,48 +93,54 @@ function (indicator_hg::IndicatorHennemannGassner)(u::AbstractArray{<:Any,5},
     alpha[element] = min(alpha_max, alpha_element)
   end
 
-  if (alpha_smooth)
-    # Diffuse alpha values by setting each alpha to at least 50% of neighboring elements' alpha
-    # Copy alpha values such that smoothing is indpedenent of the element access order
-    alpha_tmp .= alpha
-
-    # Loop over interfaces
-    for interface in eachinterface(dg, cache)
-      # Get neighboring element ids
-      left  = cache.interfaces.neighbor_ids[1, interface]
-      right = cache.interfaces.neighbor_ids[2, interface]
-
-      # Apply smoothing
-      alpha[left]  = max(alpha_tmp[left],  0.5 * alpha_tmp[right], alpha[left])
-      alpha[right] = max(alpha_tmp[right], 0.5 * alpha_tmp[left],  alpha[right])
-    end
-
-    # Loop over L2 mortars
-    for mortar in eachmortar(dg, cache)
-      # Get neighboring element ids
-      lower_left  = cache.mortars.neighbor_ids[1, mortar]
-      lower_right = cache.mortars.neighbor_ids[2, mortar]
-      upper_left  = cache.mortars.neighbor_ids[3, mortar]
-      upper_right = cache.mortars.neighbor_ids[4, mortar]
-      large       = cache.mortars.neighbor_ids[5, mortar]
-
-      # Apply smoothing
-      alpha[lower_left]  = max(alpha_tmp[lower_left],  0.5 * alpha_tmp[large], alpha[lower_left])
-      alpha[lower_right] = max(alpha_tmp[lower_right], 0.5 * alpha_tmp[large], alpha[lower_right])
-      alpha[upper_left]  = max(alpha_tmp[upper_left],  0.5 * alpha_tmp[large], alpha[upper_left])
-      alpha[upper_right] = max(alpha_tmp[upper_right], 0.5 * alpha_tmp[large], alpha[upper_right])
-
-      alpha[large] = max(alpha_tmp[large], 0.5 * alpha_tmp[lower_left],  alpha[large])
-      alpha[large] = max(alpha_tmp[large], 0.5 * alpha_tmp[lower_right], alpha[large])
-      alpha[large] = max(alpha_tmp[large], 0.5 * alpha_tmp[upper_left],  alpha[large])
-      alpha[large] = max(alpha_tmp[large], 0.5 * alpha_tmp[upper_right], alpha[large])
-    end
+  if alpha_smooth
+    apply_smoothing_3d!(mesh, alpha, alpha_tmp, dg, cache)
   end
 
   return alpha
 end
 
 
+function apply_smoothing_3d!(mesh::Union{TreeMesh{3}, P4estMesh{3}}, alpha, alpha_tmp, dg, cache)
```

Comment:
```suggestion
function apply_smoothing!(mesh::Union{TreeMesh{3}, P4estMesh{3}}, alpha, alpha_tmp, dg, cache)
```
The dimension is part of the `mesh` type, so you should use dispatch instead of choosing different function names. This should be adapted in 2D etc, too